crates/topos/src/components/node/services/process.rs

```diff
@@ -0,0 +1,137 @@
+    TceFailure,
+    #[error("Failure on the Sequencer")]
+    SequencerFailure,
+    #[error("Failure on the Edge")]
+    EdgeTerminated(#[from] std::io::Error),
+}
+
+pub fn generate_edge_config(
+    edge_path: PathBuf,
+    config_path: PathBuf,
+) -> JoinHandle<Result<(), Errors>> {
+    // Create the Polygon Edge config
+    info!("Generating the configuration at {config_path:?}");
+    info!("Polygon-edge binary located at: {edge_path:?}");
+    spawn(async move {
+        match CommandConfig::new(edge_path)
+            .init(&config_path)
+            .spawn()
+            .await
+        {
+            Ok(status) => {
+                info!("Edge process terminated: {status:?}");
+                Ok(())
+            }
+            Err(e) => {
+                println!("Failed to run the edge binary: {e:?}");
+                Err(Errors::EdgeTerminated(e))
+            }
+        }
+    })
+}
+
+pub(crate) fn spawn_sequencer_process(
+    config: SequencerConfig,
+    keys: &SecretManager,
+    shutdown: (CancellationToken, mpsc::Sender<()>),
+) -> JoinHandle<Result<(), Errors>> {
+    let config = SequencerConfiguration {
+        subnet_id: config.subnet_id,
+        public_key: keys.validator_pubkey(),
+        subnet_jsonrpc_http: config.subnet_jsonrpc_http,
+        subnet_jsonrpc_ws: config.subnet_jsonrpc_ws,
+        subnet_contract_address: config.subnet_contract_address,
+        tce_grpc_endpoint: config.tce_grpc_endpoint,
+        signing_key: keys.validator.clone().unwrap(),
+        verifier: 0,
+        start_block: config.start_block,
+    };
+
+    debug!("Sequencer args: {config:?}");
+    spawn(async move {
+        topos_sequencer::run(config, shutdown).await.map_err(|e| {
+            error!("Failure on the Sequencer: {e:?}");
+            Errors::SequencerFailure
+        })
+    })
+}
+
+pub(crate) fn spawn_tce_process(
+    config: TceConfig,
+    keys: SecretManager,
+    genesis: Genesis,
+    shutdown: (CancellationToken, mpsc::Sender<()>),
+) -> JoinHandle<Result<(), Errors>> {
+    let tce_config = TceConfiguration {
+        boot_peers: genesis
+            .boot_peers(Some(topos_p2p::constants::TCE_BOOTNODE_PORT))
+            .into_iter()
+            .chain(config.parse_boot_peers())
+            .collect::<Vec<_>>(),
+        validators: genesis.validators().expect("Cannot parse validators"),
+        auth_key: keys.network.map(AuthKey::PrivateKey),
+        signing_key: keys.validator.map(AuthKey::PrivateKey),
+        tce_addr: format!("/ip4/{}", config.libp2p_api_addr.ip()),
+        tce_local_port: config.libp2p_api_addr.port(),
+        tce_params: ReliableBroadcastParams::new(genesis.validator_count()),
+        api_addr: config.grpc_api_addr,
+        graphql_api_addr: config.graphql_api_addr,
+        metrics_api_addr: config.metrics_api_addr,
+        storage: StorageConfiguration::RocksDB(Some(config.db_path)),
+        network_bootstrap_timeout: Duration::from_secs(180),
```

Comment:
180 sec is pretty long!

Reply:
Ping @hadjiszs. Was there a specific reason for `180`?

Reply:
Hm I'm not sure, if I recall correctly this timeout was set empirically because of the time it takes in some tests that involve docker and docker compose but I'm not entirely sure. @Freyskeyd probably knows more about it

We can try reducing it and see if it breaks any existing tests

Reply:
Ok, let's do 90 and see what happens?